src/network/networkd-state-file.c

```diff
@@ -601,7 +601,7 @@ static void serialize_resolvers(
 
                 r = sd_dhcp_lease_get_dnr(lease, &resolvers);
                 if (r < 0)
-                        return (void) log_warning_errno(r, "Failed to get DNR from DHCP lease, ignoring.");
```

Comment:
hmm, i'd prefer to keep it, but downgrade to debug level.

and while you are at it, maybe suffix the log message with `, ignoring: %m` – so that we can see the error msg

Reply:
It could be I'm missing something but as far as I can see this event is hit when everything is normal (and networkd doesn't complain about router advertisements with no options for example) so I'm not sure what it is about. I agree that there are places where it probably makes sense to complain about bogus/malfromed options but this place isn't where it's logged as far as I can tell.

Reply:
Apart from being annoying this log message appears to indicate a bug in the sense that it prevents DNRs received over DHCPv6 from being serialized. I don't understand the logic behind it either. For example if captive portals aren't received networkd just silently ignores that by explicitly ignoring `-ENODATA` in https://github.com/systemd/systemd/blob/cf95ad41b0b509a2b3c5efc3f4266d9fd494108d/src/network/networkd-dhcp-common.c#L274-L278

According to https://www.rfc-editor.org/rfc/rfc9463.html#section-3.2 there is nothing wrong with using those options coming from different sources (and resolved actually consumes them even though DNRs sometimes aren't serialized if they come via router advertisements only for example). I have to admit I'm completely puzzled.

Reply:
Looks I'm talking about https://github.com/systemd/systemd/blob/cf95ad41b0b509a2b3c5efc3f4266d9fd494108d/src/network/networkd-json.c#L1384-L1386 and it's different from this part of the code.

Reply:
#35020